### ISSUE TITLE
Fix return error type and cause for ChangeMessageVisibility request with invalid receipt handle

### ISSUE THREAD

**Issue by split-hub** [localstack/localstack#3619]
# Type of request: This is a ...

[X] bug report
[ ] feature request

# Detailed description
When you send a ChangeMessageVisibility request with an invalid receipt handle to SQS you should receive an error 400 with a ReceiptHandleIsInvalidException but currently localstack only returns the 400 with no error message/cause.
...

## Expected behavior
when doing the following call to localstack, a 400 error with the cause ReceiptHandleIsInvalidException should be returned. granted the `request` contains an invalid receipt handle.

```
AmazonSQS sqs = AmazonSQSClientBuilder.standard()
                    .withEndpointConfiguration(localstack.getEndpointConfiguration(LocalStackContainer.Service.SQS))
                    .withCredentials(localstack.getDefaultCredentialsProvider()).build();
sqs.changeMessageVisibility(request);
```

...

## Actual behavior
when doing the following call to localstack, a 400 error is returned but an unkown error type and no cause is given.

```
AmazonSQS sqs = AmazonSQSClientBuilder.standard()
                    .withEndpointConfiguration(localstack.getEndpointConfiguration(LocalStackContainer.Service.SQS))
                    .withCredentials(localstack.getDefaultCredentialsProvider()).build();
sqs.changeMessageVisibility(request);
```

...

# Steps to reproduce
Create a connection to Amazon SQS and execute the following code

```
AmazonSQS sqs = AmazonSQSClientBuilder.standard()
                    .withEndpointConfiguration(localstack.getEndpointConfiguration(LocalStackContainer.Service.SQS))
                    .withCredentials(localstack.getDefaultCredentialsProvider()).build();
sqs.changeMessageVisibility("some valid queue url", "invalid receipt handle", 60);
```

## Command used to start LocalStack
`private static DockerImageName localstackImage = DockerImageName.parse("localstack/localstack:0.12.6");

    private static LocalStackContainer localstack;

    static {
        localstack = new LocalStackContainer(localstackImage).withServices(LocalStackContainer.Service.S3,
                LocalStackContainer.Service.SQS, LocalStackContainer.Service.DYNAMODB, LocalStackContainer.Service.SNS,
                LocalStackContainer.Service.CLOUDWATCH);
        localstack.start();
    }`
...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

...

**Comment by usmangani1** [comment]
@split-hub Thanks for reporting the issue.
I tried to reproduce the issue with the sample set of aws cli commands added below and I see that localstack does return the valid error message.Can you please post the full java script here so that I can check if the issue is with java client?


```
usman@Usman ~ % aws sqs create-queue --endpoint-url http://localhost:4566 --queue-name test-queue.fifo --attributes "FifoQueue=true,ContentBasedDeduplication=true"               

{
    "QueueUrl": "http://localhost:4566/000000000000/test-queue.fifo"
}
usman@Usman ~ % aws sqs send-message --endpoint-url http://localhost:4566 --queue-url http://localhost:4566/000000000000/test-queue.fifo --message-body="msg2" --message-deduplication-id="dedup" --message-group-id="group"

{
    "MD5OfMessageBody": "68b619a68d46df0f5fb66f50b5ccdde1",
    "MessageId": "783667d0-f77c-bd11-ba6e-623633622422"
}
usman@Usman ~ % aws sqs receive-message --endpoint-url http://localhost:4566 --queue-url http://localhost:4566/000000000000/test-queue.fifo --attribute-names All --message-attribute-names All 

{
    "Messages": [
        {
            "MessageId": "783667d0-f77c-bd11-ba6e-623633622422",
            "ReceiptHandle": "hmisuboixyoawcpdhniibkyrvebdywanzjqxpdsbdspchqydsosfudtoakbgnkojuycggrzpczfyakfplzucasahopmyowwmabdmbzgvwprmajivybptapnufmpcnjcumrctenqfzcxulctsppmapjtthtodbjoiziefdvulklztifuzdztdttopo",
            "MD5OfBody": "68b619a68d46df0f5fb66f50b5ccdde1",
            "Body": "msg2",
            "Attributes": {
                "SenderId": "AIDAIT2UOQQY3AUEKVGXU",
                "SentTimestamp": "1613920871841",
                "ApproximateReceiveCount": "1",
                "ApproximateFirstReceiveTimestamp": "1613920913742",
                "MessageDeduplicationId": "dedup",
                "MessageGroupId": "group"
            }
        }
    ]
}
usman@Usman ~ % aws sqs change-message-visibility --endpoint-url http://localhost:4566  --queue-url http://localhost:4566/000000000000/test-queue.fifo  --receipt-handle AQEBTpyI...t6HyQg== --visibility-timeout 36000

An error occurred (ReceiptHandleIsInvalid) when calling the ChangeMessageVisibility operation: The input receipt handle is invalid.
```

**Comment by split-hub** [comment]
This is the request and error code received when I send to Localstack
```
{QueueUrl: http://localhost:32960/000000000000/12E718F93A3F69C1DDF6D1FD5E6F8AF7,ReceiptHandle: ImAnInvalidLockId,VisibilityTimeout: 0}

com.amazonaws.services.sqs.model.AmazonSQSException: null (Service: AmazonSQS; Status Code: 400; Error Code: 400 ; Request ID: null)
```

This is the request and error code received when I send to AWS
```
{QueueUrl: https://sqs.us-east-1.amazonaws.com/008119339527/12E718F93A3F69C1DDF6D1FD5E6F8AF7,ReceiptHandle: ImAnInvalidLockId,VisibilityTimeout: 0}

Method threw 'com.amazonaws.services.sqs.model.ReceiptHandleIsInvalidException' exception.
```

**Comment by bahui80** [comment]
@usmangani1 i am having the exact same issue. The exception is thrown but the description is null as specified above!

**Comment by rhlarora84** [comment]
I am also facing a similar issue

**Comment by thrau** [comment]
should be fixed with #6016
